localstack/services/cloudwatch/cloudwatch_starter.py

```diff
@@ -21,6 +28,86 @@ def put_composite_alarm(self):
     if not hasattr(cloudwatch_responses.CloudWatchResponse, "put_composite_alarm"):
         cloudwatch_responses.CloudWatchResponse.put_composite_alarm = put_composite_alarm
 
+    @patch(target=FakeAlarm.update_state)
+    def update_state(target, self, reason, reason_data, state_value):
+        old_state = self.state_value
+        target(self, reason, reason_data, state_value)
+
+        # check the state and trigger required actions
+        if self.actions_enabled and old_state != self.state_value:
```

Comment:
nitpick: we could apply the _Return Early_ pattern here, and check for conditions to return from the function (can help reduce the level of nesting):
```
if not self.actions_enabled or old_state == self.state_value:
    return
...
```